code/modules/reagents/reagent_containers/ampoules.dm

```diff
@@ -0,0 +1,402 @@
+	if(broken)
+		icon_state = "[base_icon_state]-b"
+	else if(!sealed)
+		icon_state = "[base_icon_state]-o"
+	else
+		icon_state = "[base_icon_state]"
+	return ..()
+
+/obj/item/reagent_containers/glass/ampoule/update_overlays()
+	. = ..()
+	if(!isnull(label_text))
+		. += "ampoule-label[(!istext(label_text) || length(label_text)) ? "-words" : null]"
+	if(dot_color && sealed && !broken)
+		var/mutable_appearance/dot = mutable_appearance('icons/obj/chemical.dmi', "ampoule-dot")
+		dot.color = color2hex(dot_color)
+		. += dot
+
+/obj/item/reagent_containers/glass/ampoule/obj_break(damage_flag)
+	if(broken)
+		return
+
+	snap()
+	broken = TRUE
+	sharpness = SHARP_EDGED
+	if(!reagents.total_volume)
+		return
+	chem_splash(get_turf(src), 1, reagents)
+
+/obj/item/reagent_containers/glass/ampoule/assume_air(datum/gas_mixture/giver)
+	return sealed ? air_contents.merge(giver) : ..()
+
+/obj/item/reagent_containers/glass/ampoule/remove_air(amount)
+	return sealed ? air_contents.remove(amount) : ..()
+
+/obj/item/reagent_containers/glass/ampoule/return_air()
+	return sealed ? air_contents : ..()
+
+/obj/item/reagent_containers/glass/ampoule/attack_self(mob/living/user)
+	if(!sealed)
+		return ..()
+
+	var/clean = TRUE
+	if(prob(50))
+		clean = FALSE
+	else if(HAS_TRAIT(user, TRAIT_CLUMSY) && prob(50))
+		clean = FALSE
+	else if(HAS_TRAIT(user, TRAIT_CHUNKYFINGERS) && prob(30))
+		clean = FALSE
+	else if(iscarbon(user))
+		var/mob/living/carbon/carbon_user = user
+		if(prob(carbon_user.jitteriness))
+			clean = FALSE
+
+	if(clean)
+		snap()
+	else
+		obj_break()
+	return TRUE
+
+/obj/item/reagent_containers/glass/ampoule/welder_act(mob/living/user, obj/item/tool)
+	if(sealed || user.combat_mode)
+		return ..()
+	var/heat = tool.get_temperature()
+	if(heat)
+		reagents.expose_temperature(heat)
+	try_weld_shut(tool, user)
+	return TRUE
+
+/obj/item/reagent_containers/glass/ampoule/wirecutter_act(mob/living/user, obj/item/tool)
+	if(!sealed || user.combat_mode)
+		return ..()
+	var/heat = tool.get_temperature()
+	if(heat)
+		reagents.expose_temperature(heat, coeff = 0.01)
+	try_cut_open(tool, user)
+	return TRUE
+
+/obj/item/reagent_containers/glass/ampoule/attackby(obj/item/tool, mob/living/user, params)
+	if(!sealed || !tool.force || !tool.sharpness || user.combat_mode)
+		return ..()
+	var/heat = tool.get_temperature()
+	if(heat)
+		reagents.expose_temperature(heat, coeff = 0.01)
+	try_cut_open(tool, user)
+	return TRUE
+
+/obj/item/reagent_containers/glass/ampoule/try_splash(mob/user, atom/target)
+	if(sealed)
+		if(prob(20))
+			snap()
+		else
+			obj_break()
+	return ..()
+
+/obj/item/reagent_containers/glass/ampoule/process(delta_time)
+	var/datum/gas_mixture/env = loc?.return_air()
+	if(env)
+		air_contents.archive()
+		air_contents.temperature_share(env)
+
+	air_contents.react(src)
+	reagents.expose_temperature(air_contents.temperature)
+	if(air_contents.return_pressure() > TANK_LEAK_PRESSURE)
+		obj_break(BOMB)
```

Comment:
A: You're mutating the gasmix of what might be a turf. You need to add it to active processing if it is actually a turf.

B: How many of these will exist? Is this going to effect things?

Reply:
*affect

Reply:
Removed the bit that involves sharing temp with turfs.